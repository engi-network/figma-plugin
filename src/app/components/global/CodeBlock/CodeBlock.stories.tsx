import CodeBlock from './CodeBlock'

export default {
  component: CodeBlock,
  title: 'Global/Components/CodeBlock',
}

const codeString = `
  function Button({
    backgroundColor,
    primary = false,
    size = BUTTON_SIZE.MEDIUM,
    className,
    onClick,
    children,
    disabled,
    ...rest
  }: ButtonProps) {
    const mode = primary ? styles.primary : styles.secondary

    return (
      <button
        type="button"
        css={[
          styles.button,
          styles[size],
          mode,
          disabled && styles.disabled,
          backgroundColor && { background: backgroundColor },
        ]}
        className={className}
        onClick={onClick}
        disabled={disabled}
        {...rest}
      >
        {children}
      </button>
    )
  }
`
export function CodeBlockStory() {
  return (
    <div className="bg-slate-800 h-full p-10">
      <CodeBlock codeString={codeString} className={'w-36 h-20'} />
    </div>
  )
}
