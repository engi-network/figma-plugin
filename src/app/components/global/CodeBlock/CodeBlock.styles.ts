import { StylesMap } from '~/app/lib/constants'

const styles: StylesMap = {
  highlighter: {
    background: 'transparent',
    fontSize: 10,
    fontWeight: 500,
    height: '100%',
    lineHeight: '15px',
    overflow: 'hidden',
    padding: 0,
  },
}

export default styles
