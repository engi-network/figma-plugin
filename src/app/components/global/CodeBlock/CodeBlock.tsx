import cn from 'classnames'
import {
  Light as SyntaxHighlighter,
  SyntaxHighlighterProps,
} from 'react-syntax-highlighter'
import js from 'react-syntax-highlighter/dist/esm/languages/hljs/javascript'
import { atomOneDark } from 'react-syntax-highlighter/dist/esm/styles/hljs'

import styles from './CodeBlock.styles'

SyntaxHighlighter.registerLanguage('javascript', js)

export interface CodeBlockProps extends SyntaxHighlighterProps {
  className?: string
  codeString: string
  showLineNumbers?: boolean
}

function CodeBlock({
  className,
  codeString,
  showLineNumbers,
  customStyle,
  ...rest
}: CodeBlockProps) {
  const rootClasses = cn(
    className,
    'border border-primary-gray/30 overflow-hidden p-[10px]',
  )

  return (
    <div className={rootClasses}>
      <SyntaxHighlighter
        language="javascript"
        style={atomOneDark}
        wrapLines
        // wrapLongLines
        showLineNumbers={showLineNumbers}
        customStyle={{ ...styles.highlighter, ...customStyle }}
        lineProps={(lineNumber) => {
          const style = {} as Record<string, string>
          if (lineNumber > 3) {
            style.display = 'none'
          }
          return { style }
        }}
        {...rest}
      >
        {codeString}
      </SyntaxHighlighter>
    </div>
  )
}

export default CodeBlock
