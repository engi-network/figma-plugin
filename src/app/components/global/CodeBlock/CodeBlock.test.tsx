import renderer from 'react-test-renderer'

import CodeBlock from './CodeBlock'

const codeString = `const TestCompenent = () => {
  return <div>Test</div>
}
`

describe('CodeBlock', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(<CodeBlock codeString={codeString} className={'w-36 h-20'} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
