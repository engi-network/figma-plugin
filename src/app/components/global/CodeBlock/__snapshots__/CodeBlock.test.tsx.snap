// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeBlock should render correctly 1`] = `
<div
  className="w-36 h-20 border border-primary-gray/30 overflow-hidden p-[10px]"
>
  <pre
    style={
      Object {
        "background": "transparent",
        "color": "#000",
        "display": "block",
        "fontSize": 10,
        "fontWeight": 500,
        "height": "100%",
        "lineHeight": "15px",
        "overflow": "hidden",
        "overflowX": "auto",
        "padding": 0,
      }
    }
  >
    <code
      className="language-javascript"
      style={
        Object {
          "whiteSpace": "pre-wrap",
        }
      }
    >
      <span
        style={Object {}}
      >
        <span
          style={
            Object {
              "color": "#954121",
            }
          }
        >
          const
        </span>
        <span
          style={Object {}}
        >
           TestCompenent = 
        </span>
        <span
          className="hljs-function"
          style={Object {}}
        >
          () =&gt;
        </span>
        <span
          style={Object {}}
        >
           {

        </span>
      </span>
      <span
        style={Object {}}
      >
        <span
          style={Object {}}
        >
            
        </span>
        <span
          style={
            Object {
              "color": "#954121",
            }
          }
        >
          return
        </span>
        <span
          style={Object {}}
        >
           
        </span>
        <span
          className="xml"
          style={
            Object {
              "color": "#000080",
              "fontWeight": "normal",
            }
          }
        >
          &lt;
        </span>
        <span
          className="xml"
          style={
            Object {
              "color": "#000080",
              "fontWeight": "normal",
            }
          }
        >
          div
        </span>
        <span
          className="xml"
          style={
            Object {
              "color": "#000080",
              "fontWeight": "normal",
            }
          }
        >
          &gt;
        </span>
        <span
          className="xml"
          style={Object {}}
        >
          Test
        </span>
        <span
          className="xml"
          style={
            Object {
              "color": "#000080",
              "fontWeight": "normal",
            }
          }
        >
          &lt;/
        </span>
        <span
          className="xml"
          style={
            Object {
              "color": "#000080",
              "fontWeight": "normal",
            }
          }
        >
          div
        </span>
        <span
          className="xml"
          style={
            Object {
              "color": "#000080",
              "fontWeight": "normal",
            }
          }
        >
          &gt;
        </span>
        <span
          style={Object {}}
        >
          

        </span>
      </span>
      <span
        style={Object {}}
      >
        }

      </span>
      <span
        style={Object {}}
      >
        
      </span>
    </code>
  </pre>
</div>
`;
